Rapr/UpdateManager.cs

```diff
@@ -2,36 +2,63 @@
 using System.Net;
 using System.Net.Http;
 using System.Net.Http.Headers;
-
+using System.Threading.Tasks;
 using Newtonsoft.Json.Linq;
 
 namespace Rapr
 {
-    public static class UpdateManager
+    public class UpdateManager : IUpdateManager, IDisposable
     {
-        public static (Version version, string pageUrl, string downloadUrl) GetLatestVersionInfo()
+        private const string versionInfoUrl = "https://api.github.com/repos/lostindark/DriverStoreExplorer/releases/latest";
+        private readonly HttpClient httpClient = new HttpClient();
+        private bool disposedValue = false; // To detect redundant calls
+
+        public async Task<VersionInfo> GetLatestVersionInfo()
         {
             ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
 
-            using (HttpClient httpClient = new HttpClient())
+            httpClient.DefaultRequestHeaders
+                .Accept
+                .Add(new MediaTypeWithQualityHeaderValue("application/vnd.github.v3+json"));
+
+            httpClient.DefaultRequestHeaders
+                .Add("User-Agent", "System.Net.Http Agent");
+
+            var response = await httpClient.GetAsync(new Uri(versionInfoUrl)).ConfigureAwait(false);
+
+            if (response.IsSuccessStatusCode)
+            {
+                var releaseInfo = JObject.Parse(response.Content.ReadAsStringAsync().GetAwaiter().GetResult());
```

Comment:
Do not call .GetAwaiter().GetResult() inside async method.

Reply:
Addressed.###